heartbeat/smb-share.in

```diff
@@ -0,0 +1,494 @@
+            <longdesc lang="en">
+            This specifies a UNIX user name that will be assigned as the default user for all users connecting to this service. This is useful for sharing files.
+            </longdesc>
+            <shortdesc lang="en">force user parameter</shortdesc>
+            <content type="string" default="${OCF_RESKEY_force_user_default}" />
+        </parameter>
+        
+        <parameter name="force_group" unique="0" required="0">
+            <longdesc lang="en">
+            This specifies a UNIX group name that will be assigned as the default primary group for all users connecting to this service.
+            This is useful for sharing files by ensuring that all access to files on service will use the named group for their permissions checking.
+            </longdesc>
+            <shortdesc lang="en">force group parameter</shortdesc>
+            <content type="string" default="${OCF_RESKEY_force_group_default}" />
+        </parameter>
+        
+        <parameter name="extraopt" unique="0" required="0">
+            <longdesc lang="en">
+            This option can be used to define an additional key = value pair.
+            In this parameter also a semicolon could be placed.
+            Need to set e.g somthinspecial = value
+            </longdesc>
+            <shortdesc lang="en">additional key value pair</shortdesc>
+            <content type="string" default="${OCF_RESKEY_extraopt_default}" />
+        </parameter>
+        
+        <parameter name="extraopt_list" unique="0" required="0">
+            <longdesc lang="en">
+            This option can be used to define multiple additional key = value pairs.
+            Define the list of element semicolon separated.
+            e.g somethingspecial = value;one more = value2
+            </longdesc>
+            <shortdesc lang="en">additional key value pairs as semicolon separated list</shortdesc>
+            <content type="string" default="${OCF_RESKEY_extraopt_list_default}" />
+        </parameter>
+        
+        <parameter name="outfile" unique="1" required="1">
+            <longdesc lang="en">
+            Set the path and filename where the snipped should be written.
+            example "/etc/samba/conf.d/myshare.inc".
+            </longdesc>
+            <shortdesc lang="en">outputfile</shortdesc>
+            <content type="string" default="${OCF_RESKEY_outfile_default}" />
+        </parameter>
+        
+        <parameter name="confd" unique="0" required="0">
+            <longdesc lang="en">
+            Set the path where the includes will be written. This folder have to be a tmpfs mount
+            This defaults to "/etc/samba/conf.d".
+            </longdesc>
+            <shortdesc lang="en">confd directory</shortdesc>
+            <content type="string" default="${OCF_RESKEY_confd_default}" />
+        </parameter>
+        
+        <parameter name="includesfile" unique="1" required="0">
```

Comment:
Why shouldn't two `smb-share` resources be allowed to use the same `includesfile`?

Reply:
Because the agent is meant for one share per file.

Reply:
The `includesfile` isn't the same as the `outfile`. There's one share per `outfile`, but multiple outfiles could be listed in the same `includesfile`. For example, by default, all of the includes get appended to the default `includesfile`: `/etc/samba/conf.d/pacemaker-includes.conf`. So I see no reason why a user shouldn't be able to specify some different `includesfile` for multiple `smb-share` resources -- e.g., `includesfile=/etc/samba/conf.d/custom-includes.conf` on two separate resources.

See the addinclude function below:
```
smb_share_addinclude() {
        if [ ! -e ${OCF_RESKEY_includesfile} ];then
            echo '[global]' >  ${OCF_RESKEY_includesfile}
        fi
        if [ $(grep -c "include = $OCF_RESKEY_outfile" ${OCF_RESKEY_includesfile}) -eq 0 ];then
            echo "include = $OCF_RESKEY_outfile" >> ${OCF_RESKEY_includesfile}
        fi
}
```

Reply:
Ah. So it just deletes outfile.

We'll have to delete the include file during the stop-action.

Reply:
No, because multiple `smb-share` resources may use the same `includesfile`. By default, they do. We don't want to delete the `includesfile` when when `smb-share` stops. Another one may still be using it.

The issue I'm raising is that even though by default all `smb-share` resources use the same `includesfile`, the user won't be allowed to *specify* the same `includesfile` for multiple resources.

Reply:
The stop operation deletes the `include = <outfile>` line from the `includesfile`; it doesn't delete the `includesfile`, nor should it.

Reply:
Hello @nrwahl2,

yes you pointed out a correct mistake.
The includesfile shouldn't be uniq. 
In my setup I used the default, so the problem wasn't shown.
I will fix it immediatly.
Thanks for pointing to it!

Reply:
Thanks, @tobias-d-oe!

Reply:
done